### DISCUSSION TITLE
Possible bug in EntityManager::transactional

### DISCUSSION THREAD

**Discussion by ghost** [doctrine/doctrine2#9217] [Support Questions] [answered]
See:

https://github.com/doctrine/orm/blob/15ec77fa79130ad5bbcb20f0630bc43d48c0436d/lib/Doctrine/ORM/EntityManager.php#L240

If the callback returns 0 (or something PHP evaluates falsy), EntityManager::transactional evaluates it as false and returns true.

Why would returning zero be not valid return value?

**Comment by yethee** [comment, marked as answer]
Since 2.10.0 version you can use [`EntityManager::wrapInTransaction()`](https://github.com/doctrine/orm/pull/8419)

**Reply by ghost** [reply]
Seems the issue is at least mentioned there. Thanks!
